ambry-server/src/test/java/com.github.ambry.server/AmbryServerRequestsTest.java

```diff
@@ -183,6 +191,61 @@ public void after() throws InterruptedException {
     storageManager.shutdown();
   }
 
+  /**
+   * Tests that requests are validated based on local store state.
+   */
+  @Test
+  public void validateRequestsTest() {
+    // choose several replicas and make them in different states (there are 10 replicas on current node)
+    List<ReplicaId> localReplicas = clusterMap.getReplicaIds(dataNodeId);
+    Map<ReplicaState, ReplicaId> stateToReplica = new HashMap<>();
+    int cnt = 0;
+    for (ReplicaState state : EnumSet.complementOf(EnumSet.of(ReplicaState.ERROR))) {
+      stateToReplica.put(state, localReplicas.get(cnt++));
```

Comment:
nit: I would discourage using ++ in the same place where you use cnt. This would cause confusion. Can you do
```
stateToReplica.put(state, localReplicas.get(cnt));
cnt++;
```

Reply:
fixed